src/components/common/EffectiveDate.vue

```diff
@@ -0,0 +1,198 @@
+    return this.getCurrentDate
+  }
+
+  /** Called when component is mounted. */
+  mounted (): void {
+    // load initial data
+    this.dateFormatted = this.formatDate(this.initialEffectiveDate)
+  }
+
+  /**
+   * Local helper to change date from YYYY-MM-DD to YYYY/MM/DD.
+   * @returns The formatted date.
+   */
+  formatDate (date: string, validate = true): string {
+    if (validate && !this.isValidDate(date, '-')) return ''
+    const [year, month, day] = date.split('-')
+    return `${year}/${month}/${day}`
+  }
+
+  /**
+   * Local helper to change date from YYYY/MM/DD to YYYY-MM-DD.
+   * @returns The parsed date.
+   */
+  parseDate (date: string): string {
+    // changes date from YYYY/MM/DD to YYYY-MM-DD
+    if (!this.isValidDate(date, '/')) return ''
+    const [year, month, day] = date.split('/')
+    return `${year}-${month}-${day}`
+  }
+
+  /**
+   * Local helper to determine if passed-in date is valid.
+   * @returns True if date is valid, otherwise false.
+   */
+  isValidDate (date, separator): boolean {
+    return (isNotNull.call(this, date) &&
+      isValidFormat.call(this, date, separator) &&
+      isValidEffectiveDate.call(this, date, separator))
+  }
+
+  /**
+   * When prop changes, load (initial) data.
+   */
+  @Watch('initialEffectiveDate')
+  onInitialEffectiveDateChanged (val: string): void {
+    if (val) {
+      this.dateFormatted = this.formatDate(val)
+    }
+  }
+
+  /**
+   * When text field changes, update date picker.
+   */
+  @Watch('dateFormatted')
+  onDateFormattedChanged (val: string): void {
+    this.date = this.parseDate(val)
+  }
+
+  /**
+   * When date picker changes, update text field etc.
+   */
+  @Watch('date')
+  onDateChanged (val: string): void {
+    const effectiveDate = this.isValidDate(val, '-') ? val : null
+    // only update text field if date is valid
+    // this is to retain previous invalid values
+    if (effectiveDate) {
+      this.dateFormatted = this.formatDate(val)
+    }
+    this.emitEffectiveDate(effectiveDate)
+    this.emitValid(Boolean(effectiveDate))
+  }
+
+  /** Validate date text field */
+  @Watch('validateForm')
+  validateDateField (): void {
+    if (this.validateForm && !this.dateFormatted) {
+      this.$refs.textarea.validate()
+      this.$refs.textarea.error = true
+    }
+  }
+
+  /**
+   * Emits an event with the new value of Effective Date.
+   */
+  @Emit('update:effectiveDate')
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  emitEffectiveDate (val: string): void { /* no empty function */ }
+
+  /**
+   * Emits an event indicating whether or not this component is valid.
+   */
+  @Emit('valid')
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  emitValid (val: boolean): void { /* no empty function */ }
+}
+</script>
+
+<style lang="scss" scoped>
+@import '@/assets/styles/theme.scss';
+.title-label {
+  color: $gray9;
+  font-weight: bold;
+}
+
+.validationErrorInfo {
+  color: red;
```

Comment:
We use a specific red for errors -- see `$app-red` (aka `error` if used in a Vuetify component).

(Ask if you're wondering about the differences ^^ .)